src/proxy/inbound.rs

```diff
@@ -94,12 +98,35 @@ impl Inbound {
         let stream = crate::hyper_util::tls_server(acceptor, self.listener);
         let mut stream = stream.take_until(Box::pin(self.drain.signaled()));
 
-        let (sub_drain_signal, sub_drain) = drain::channel();
+        // spawn a task which subscribes to watch updates and asserts rbac against this proxy's connections, closing the ones which have become denied
+        let (stop_tx, mut stop_rx) = watch::channel(());
+        let state = self.state.clone();
+        let connection_manager = self.connection_manager.clone();
+
+        tokio::spawn(async move {
+            let mut policies_changed = state.read().policies.subscribe();
+            loop {
+                tokio::select! {
+                    _ = stop_rx.changed() => {
+                        return
+                    }
+                    _ = policies_changed.changed() => {
+                        let connections = connection_manager.connections().await;
+                        for conn in connections {
+                            if !state.assert_rbac(&conn).await {
+                                connection_manager.drain(&conn).await;
```

Comment:
This definitely needs some nice logging since it will be a pretty major "WTF just happened" in some cases

Reply:
firm agree. TY